library/std/src/thread/mod.rs

```diff
@@ -698,17 +698,20 @@ where
 }
 
 thread_local! {
+    // Invariant: `CURRENT` and `CURRENT_ID` will always be initialized together.
```

Comment:
Could you also add a note about `CURRENT_ID` always holding the same ID as the thread in `CURRENT`, if it is initialized?